src/qamqpclient.cpp

```diff
@@ -687,8 +697,10 @@ QAmqpQueue *QAmqpClient::createQueue(const QString &name, int channelNumber)
     connect(this, SIGNAL(disconnected()), queue, SLOT(_q_disconnected()));
     queue->d_func()->open();
 
-    if (!name.isEmpty())
+    if (!name.isEmpty()) {
         queue->setName(name);
+        d->queues[name] = queue;
```

Comment:
On 09/04/15 22:36, Matt Broadstone wrote:
> In src/qamqpclient.cpp
> <https://github.com/mbroadst/qamqp/pull/26#discussion_r28057401>:
> 
>>          queue->setName(name);
>> +        d->queues[name] = queue;
> 
> same here?

Slightly different case here.  A request for the '' exchange is a
specific exchange, the default one.  In that case you almost always want
to return the reference to that same exchange.

That was an oversight on my part, and one I'll correct in commit
3a7c6f30f5661323f17c658a6777b9725a017940.

As for queues, a nameless queue gets assigned a name by the AMQP broker,
and so while the queue will be initially nameless, that will change once
queue.declare-ok is received.

You are correct though: I should perhaps listen for a 'declared' signal
from the queue and have the client add the queue with its new name.
-- 
Stuart Longland (aka Redhatter, VK4MSL)

I haven't lost my mind...
  ...it's backed up on a tape somewhere.